File: Array/C++/Question1/Solution_3.cpp
// 1. Find the Maximum and Minimum Element in an Array

// Using algorithm library (sort)

// directly access the first and last elements

#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int arr[] = {3, 7, 2, 5, 9, 1};
    int n = sizeof(arr) / sizeof(arr[0]);

    sort(arr, arr + n);

    cout << "Maximum: " << arr[n - 1] << endl;
    cout << "Minimum: " << arr[0] << endl;

    return 0;
}

File: Array/Javascript/Question1/Solution3.js
function findMaxMin(arr) {
    if (arr.length === 0) return { max: null, min: null };
    
    const sortedArr = [...arr].sort((a, b) => a - b);
    return { max: sortedArr[sortedArr.length - 1], min: sortedArr[0] };
}

const arr = [3, 5, 7, 2, 8, 4];
console.log(findMaxMin(arr));  
